x/gov/genesis.go

```diff
@@ -6,19 +6,37 @@ import (
 
 // GenesisState - all staking state that must be provided at genesis
 type GenesisState struct {
-	StartingProposalID int64 `json:"starting_proposalID"`
+	StartingProposalID int64             `json:"starting_proposalID"`
+	DepositProcedure   DepositProcedure  `json:"deposit_period"`
+	VotingProcedure    VotingProcedure   `json:"voting_period"`
+	TallyingProcedure  TallyingProcedure `json:"tallying_procedure"`
 }
 
-func NewGenesisState(startingProposalID int64) GenesisState {
+func NewGenesisState(startingProposalID int64, dp DepositProcedure, vp VotingProcedure, tp TallyingProcedure) GenesisState {
 	return GenesisState{
 		StartingProposalID: startingProposalID,
+		DepositProcedure:   dp,
+		VotingProcedure:    vp,
+		TallyingProcedure:  tp,
 	}
 }
 
 // get raw genesis raw message for testing
 func DefaultGenesisState() GenesisState {
 	return GenesisState{
 		StartingProposalID: 1,
+		DepositProcedure: DepositProcedure{
+			MinDeposit:       sdk.Coins{sdk.NewCoin("steak", 10)},
+			MaxDepositPeriod: 200,
+		},
+		VotingProcedure: VotingProcedure{
+			VotingPeriod: 200,
+		},
+		TallyingProcedure: TallyingProcedure{
+			Threshold:         sdk.NewRat(1, 2),
+			Veto:              sdk.NewRat(1, 3),
+			GovernancePenalty: sdk.NewRat(1, 100),
+		},
```

Comment:
Is there any other point to splitting up the governance parameters into these `Procedure` structs besides organizational? If not I'd almost prefer to see all the parameters in one struct - seems a bit confusing to add another layer of abstraction where it's not needed by the code

Reply:
Voting params, DepositParams, and TallyingParams are used at different times.  No need to Unmarshal the voting params when I'm just worried about tallying.  Also, in future, there will be different types of "tallying params" and stuff for different types of proposals

Reply:
Interesting so in the future tallying params should actually be an `interface` as opposed to a `struct` to allow for different parameters - then functions such as `CheckVeto` could be functions on `TallyingProceedure`

Reply:
👍  seems fine to leave as is for now, thanks for the insight